modules/travel_pay/app/services/travel_pay/appointments_service.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module TravelPay
+  class AppointmentsService
+    def get_appointments_by_date(veis_token, btsss_token, params = {})
+      faraday_response = client.get_all_appointments(veis_token, btsss_token, { 'excludeWithClaims' => true })
+      raw_appointments = faraday_response.body['data'].deep_dup
+
+      appointments = filter_by_date(params['appt_datetime'], raw_appointments)
+
+      {
+        data: appointments || []
+      }
+    end
+
+    private
+
+    def filter_by_date(date_string, appointments)
+      if date_string.present?
+        parsed_appt_date = Date.parse(date_string)
```

Comment:
@kjduensing I wondered about parsing this date string here.... We might need to do an exact match, not just by date, but also time if they have multiple appointments on the same day, to make sure we're matching the correct VAOS appt to the correct BTSSS appointment.

Reply:
I would recommend matching on the exact time. The information is allegedly coming from the same data source, so matching wouldn't be an issue.

The reason we're making this call is to get a single appointment so we can attach a claim to it. To me, it makes more sense to try an find a specific BTSSS appointment. This would also potentially change the return type from an array of appointments to a single appointment object.